### DISCUSSION TITLE
New report type: webhook

### DISCUSSION THREAD

**Discussion by kistoth90** [apache/superset#29928] [Show and tell]
Hi,

I made some upgrade on the Superset: I add **webhook** as selectable report type. 
![Képernyőkép ekkor: 2024-08-13 11-23-39](https://github.com/user-attachments/assets/6e075d76-0605-4729-9bc4-24e0592aab53)

![Képernyőkép ekkor: 2024-08-13 11-24-05](https://github.com/user-attachments/assets/cbadedc9-052c-460f-b05b-3779aaa20069)

By the use of webhook the user can post csv/png/pdf to any endpoint, with some informative parameters as:
- name of the report
- id of the report
- report type (chart, or dashboard)
- content id
- content format

For security reasons I made a new optional parameter: WEBHOOK_SECRET. It's a secret string between the Superset and the webhook endpoint. In the case where WEBHOOK_SECRET is filled, Superset adds an "X-Webhook-Signature" parameter to the header of the post call, which hashes the json data to be sent. The receiving party can verify that the party sending the webhook is the real sender by hashing the received json data and comparing it with the sending party's "X-Webhook-Signature" parameter. If the two parameters do not match, the receiving party may reject the call because the sender is not the supposed host.

![Képernyőkép ekkor: 2024-08-13 12-04-57](https://github.com/user-attachments/assets/99f1dcb2-d739-4c25-b8f5-4678e00717e1)

On the report log, we can get feedback about the successful, and not successful requests as well.

![Képernyőkép ekkor: 2024-08-13 12-27-31](https://github.com/user-attachments/assets/6041296a-07b1-4d0e-958a-8ff73078bac3)

I'm happy to share my code with you guys!

**Comment by Bindu-yadav8** [comment]
Hi @kistoth90 

I hope you're doing well.

I came across your contribution on GitHub regarding the webhook notification feature within the Superset alerting system. It aligns perfectly with a requirement I'm currently working on, where I need to implement REST API-based alerts in Superset. Specifically, I'm looking for a way to trigger notifications using webhooks when alerts are raised.

If it's not too much trouble, could you kindly share the code or provide some guidance on how you implemented this feature? I would really appreciate any advice or example code you could share, as it would be incredibly helpful for my project.

Thank you for your time, and I look forward to hearing from you.

Regards,
Bindu

**Reply by kistoth90** [reply]
Hi Bindu,

Feel free use the latest version from my repository! You find it [HERE](https://github.com/kistoth90/superset/tree/webhook)

I believe the feature is pretty straightforward, if you have any question let me know!

Regards,
Csaba

**Reply by shubham13spd** [reply]
@Bindu-yadav8 Have you tried it and is it working ??

**Reply by SachinTaneja123** [reply]
@kistoth90 The link you shared is not working, can you please reshare it again

**Comment by Ralkion** [comment]
Seems like a generic webhook approach would enable people to send the reports to whatever final destination they choose. I'd love to see this included in the main branch.

**Comment by rusackas** [comment]
Hey @kistoth90 - others were asking for this feature recently. Would you be willing to consider contributing it upstream? It has myriad use cases!
